realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -117,6 +119,17 @@ private RealmCache(RealmConfiguration config) {
 
         RefAndCount refAndCount = cache.refAndCountMap.get(RealmCacheType.valueOf(realmClass));
 
+        if (refAndCount.globalCount == 0) {
+            SharedRealm sharedRealm = SharedRealm.getInstance(configuration);
+            sharedRealm.beginTransaction();
+            if (Table.migratePrimaryKeyTableIfNeeded(sharedRealm)) {
```

Comment:
No, but this is not a migration of user data, but a migration of an internal meta table.